### PR TITLE
Manage/ajax call update

### PR DESCRIPTION
**Pull request by phuens**
Updated the Ajax call for the manage page.

### CODE REVIEW

File: lexos/static/js/scripts_manage.js [hunk 1]
```diff
@@ -1,803 +1,906 @@
...
+            i = ':eq(' + index + ')'
+            table.rows(i).select()
+            activeRows.push($(this).attr('id'))
+        }
+        if (activeRows.length != 0) {
+            $('#bttn-downloadSelectedDocs').show()
+        }
+    })
+    var numberOfFileDone = parseInt($('.fa-folder-open-o')[0].id)
+    $('.col-sm-5').append('<p style=\'display:inline; float:left; width:200px !important;\' id=\'name\'></p>')
```
Comment (chantisnake) [inline lexos/static/js/scripts_manage.js]: Add comments

File: lexos/static/js/scripts_manage.js [hunk 2]
```diff
@@ -1,803 +1,906 @@
...
+                }
+            )
     })
+        .nodes()
+        .draw()
+
+    /* Add all rows with .selected to the DataTables activeRows array
+     It does not appear that this array needs to be maintained after
+     initialisation, but the code to do so is commented out for de-bugging. */
+    activeRows = []
```
Comment (chantisnake) [inline lexos/static/js/scripts_manage.js]: Move this into a separate function `getActiveRows` or stuff like that

File: lexos/static/js/scripts_manage.js [hunk 3]
```diff
@@ -1,803 +1,906 @@
...
+    $('tbody tr').each(function (index) {
+        if ($(this).hasClass('selected')) {
```
Comment (chantisnake) [inline lexos/static/js/scripts_manage.js]: Never use `this`, use `event.currentTarget`

https://api.jquery.com/event.currentTarget/

File: lexos/static/js/scripts_manage.js [hunk 4]
```diff
@@ -1,803 +1,906 @@
...
+            i = ':eq(' + index + ')'
+            table.rows(i).select()
```
Comment (chantisnake) [inline lexos/static/js/scripts_manage.js]:
```js
table.row(`:eq(${index})`).select()
```
You don't really need the `i`

Also, add documents I don't understand what these lines are talking about

File: lexos/static/js/scripts_manage.js [hunk 5]
```diff
@@ -1,803 +1,906 @@
...
+            activeRows.push($(this).attr('id'))
+        }
+        if (activeRows.length != 0) {
+            $('#bttn-downloadSelectedDocs').show()
+        }
+    })
+    var numberOfFileDone = parseInt($('.fa-folder-open-o')[0].id)
```
Comment (chantisnake) [inline lexos/static/js/scripts_manage.js]: Always use `const`

File: lexos/static/js/scripts_manage.js [hunk 6]
```diff
@@ -1,803 +1,906 @@
...
 
-  // Area Select events callback
-  $('#demo').DataTable()
-    .on('select', function (e, dt, type, indexes) {
-      if (type === 'row') {
-        var data = $('#demo').DataTable().rows(indexes).data()[0]
-        // console.info("select", data);
-      }
+    // Handle select events
+    table
```
Comment (chantisnake) [inline lexos/static/js/scripts_manage.js]: Move to a separate function `registerSelectDeselect(table)`

File: lexos/static/js/scripts_manage.js [hunk 7]
```diff
@@ -1,803 +1,906 @@
...
+        .on('select', function (e, dt, type, indexes) {
+            // Get selected rows as a jQuery object
+            var selected_rows = table.rows(indexes).nodes().to$()
```
Comment (chantisnake) [inline lexos/static/js/scripts_manage.js]: Never use `var`, use `const` instead

File: lexos/static/js/scripts_manage.js [hunk 8]
```diff
@@ -1,803 +1,906 @@
...
+            document.getElementById('name').innerHTML = table.rows('.selected').data().length + ' active documents' // add the correct counter text to the p
+            $('#bttn-downloadSelectedDocs').show()
+        })
+        .on('deselect', function (e, dt, type, indexes) {
+            // Get deselected rows as a jQuery object
+            var deselected_rows = table.rows(indexes).nodes().to$()
+            // Call the ajax function
+            disableRows(deselected_rows)
+            handleSelectButtons(table.rows().ids().length, table.rows({selected: true}).ids().length)
+            document.getElementById('name').innerHTML = table.rows('.selected').data().length + ' active documents' // same as the other one
```
Comment (chantisnake) [inline lexos/static/js/scripts_manage.js]: Probably can use jquery

File: lexos/static/js/scripts_manage.js [hunk 9]
```diff
@@ -1,803 +1,906 @@
...
+                $('#bttn-downloadSelectedDocs').hide()
+            }
+        })
+
+    // Area Select events callback
+    $('#demo').DataTable()
+        .on('select', function (e, dt, type, indexes) {
+            if (type === 'row') {
+                var data = $('#demo').DataTable().rows(indexes).data()[0]
+                // console.info("select", data);
```
Comment (chantisnake) [inline lexos/static/js/scripts_manage.js]: No code in comments

File: lexos/static/js/scripts_manage.js [hunk 10]
```diff
@@ -1,803 +1,906 @@
...
+            if (table.rows('.selected').data().length == 0) {
+                $('#bttn-downloadSelectedDocs').hide()
+            }
+        })
+
+    // Area Select events callback
+    $('#demo').DataTable()
+        .on('select', function (e, dt, type, indexes) {
+            if (type === 'row') {
+                var data = $('#demo').DataTable().rows(indexes).data()[0]
```
Comment (chantisnake) [inline lexos/static/js/scripts_manage.js]: const

and you can probably find a better variable name than data.

File: lexos/static/js/scripts_manage.js [hunk 11]
```diff
@@ -1,803 +1,906 @@
...
+    // Refresh all options
+    $('#context-menu').find('li').removeClass('disabled')
+    $('#context-menu').find('li').find('a').removeProp('disabled')
+
+    // Comparison values
+    num_rows = table.rows().ids().length
+    num_rows_selected = table.rows({selected: true}).ids().length
+
+    // Set config options -- Numbers refer to li elements, including dividers
+    switch (true) {
```
Comment (chantisnake) [inline lexos/static/js/scripts_manage.js]: This is a unconventional switch statement. What is wrong with plain old `if` and `else if`

File: lexos/static/js/scripts_manage.js [hunk 12]
```diff
@@ -1,803 +1,906 @@
...
+        case num_rows_selected == 0: // No rows selected
+            opts = [6, 8, 9]
```
Comment (chantisnake) [inline lexos/static/js/scripts_manage.js]: Magic numbers, document them

File: lexos/static/js/scripts_manage.js [hunk 13]
```diff
@@ -1,803 +1,906 @@
...
+                    break
+                case 'apply_class_selected':
+                    selected_rows = table.rows({selected: true}).nodes().to$()
+                    applyClassSelected(cell, selected_rows)
+                    break
+                case 'delete_all_selected':
+                    selected_rows = table.rows({selected: true}).nodes().to$()
+                    deleteAllSelected(selected_rows)
+                    break
+            }
```
Comment (chantisnake) [inline lexos/static/js/scripts_manage.js]: Add a `default`

File: lexos/static/js/scripts_manage.js [hunk 14]
```diff
@@ -1,803 +1,906 @@
...
+    // Do Ajax
+    sendAjaxRequestSaveMultiple(url, data)
+        .done(
+            function (response) {
+                console.log('save multiple: ' + response)
+                row_ids = JSON.parse(response)
+                // Update the UI
+                var reloadPage = false
+                $.each(row_ids, function (i) {
+                    id = '#' + row_ids[i]
```
Comment (chantisnake) [inline lexos/static/js/scripts_manage.js]: Never use `'str'+something`, always use template literal when you need to put some value in a string

**Comment by chantisnake** [comment]
Following comment appears in https://github.com/WheatonCS/Lexos/pull/693, but should be in this pr... sorry

----

I did a quick search. Debugger exists way before 2005.

And people pay big bucks for the JavaScript debugger that is in your hand ($129 a year https://www.jetbrains.com/webstorm/buy/)

I don't want you to waste $129 per year. Therefore use it and say bye bye to console log.

**Comment by chantisnake** [comment]
Is this just the ajax call update or through refactor? There are still plenty of problems in the code, but if we just want to update to the correct ajax call, then I guess we are fine.

**Comment by mleblanc321** [comment]
>Let Weiqi decide.
right answer;  this entire effort needs some positive(+) corrections;  perhaps a refactor of the entire front-end is too much to ask for?  if so, so be it;  but i hope we can clean up sections, like this ...

**Comment by phuens** [comment]
@chantisnake would you be able to point out the mistakes in the code and anything else that I need to do if you are free.

**Comment by Weiqi97** [comment]
Since our major task at this moment is to update jquery, I will merge this pull request as the branch has been fully tested.

**Pull request merged**